lib/backend/k8s/resources/customnoderesource.go

```diff
@@ -0,0 +1,491 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"sort"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+
+	log "github.com/Sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/kubernetes"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+// K8sResourceConverter defines an interface to map between the model and the
+// annotation representation of a resource.  These are really just ID management
+// tools since the value is serialised directly from the model.
+type CustomK8sNodeResourceConverter interface {
+	// Convert the ListInterface to the node name and resource name.
+	ListInterfaceToNodeAndName(model.ListInterface) (string, string, error)
+
+	// Convert the Key to the node name and resource name.
+	KeyToNodeAndName(model.Key) (string, string, error)
+
+	// Convert the Node name and resource name to a Key.
+	NodeAndNameToKey(string, string) (model.Key, error)
+}
+
+// customK8sNodeResourceClient implements the K8sResourceClient interface and provides a generic
+// mechanism for managing per-node resources as an annotation on the Kuberenetes Node resource.
+type customK8sNodeResourceClient struct {
+	clientSet     *kubernetes.Clientset
+	resource      string
+	converter     CustomK8sNodeResourceConverter
+	annotationKey string
+}
+
+func (c *customK8sNodeResourceClient) Create(kvp *model.KVPair) (*model.KVPair, error) {
+	logContext := log.WithFields(log.Fields{
+		"Key":      kvp.Key,
+		"Value":    kvp.Value,
+		"Resource": c.resource,
+	})
+	logContext.Debug("Create per-Node resource")
+
+	// Get the names and the latest Node settings associated with the Key.
+	_, resName, node, err := c.getNamesAndNodeFromKey(kvp.Key)
+	if err != nil {
+		logContext.WithError(err).Info("Error getting current settings when creating resource.")
```

Comment:
I don't think it should be an Error.  Perhaps a Warning though - it doesn't necessarily mean anything bad is happening with our code, just that the upstream API is rejecting the Get Node query.